block/block.go

```diff
@@ -100,13 +100,13 @@ func (b *neoBlock) SetMerkleRoot(r util.Uint256) {
 }
 
 // Timestamp implements Block interface.
-func (b *neoBlock) Timestamp() uint32 {
-	return b.base.Timestamp
+func (b *neoBlock) Timestamp() uint64 {
+	return uint64(b.base.Timestamp)
```

Comment:
The user of this API expects you to return nanosecond-precision timestamp. And for some reason I think that you only have second-level precision in your `uint32`. Thus, to make a proper conversion you have to multiply here and divide down below (as you're receiving nanoseconds in `SetTimestamp` while only storing seconds).